url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/repl/ReplEngine.java

```diff
@@ -0,0 +1,475 @@
+ *    <li>A line with a <b><code>backslash</code></b> will automatically continue with the next line. This allows for some
+ *    nice formatting.</li>
+ * </ul>
+ *
+ */
+public class ReplEngine
+implements IExtensions
+{
+    private ExtensionRepositoryBuilder extensions = new ExtensionRepositoryBuilder();
+
+    public static final String INPUT = "*input";
+    public static final String OUTPUT = "*output";
+    public static final String ERROR = "*error";
+
+    private IContext context;
+    private IEval eval;
+    private StringBuilder currCmd = new StringBuilder();
+    private Parser parser = new Parser();
+    private static final String PROMPT_NORMAL = "> ";
+    private static final String PROMPT_CONTINUE = "+ > ";
+    private String prompt = "> ";
+    private LineNumberReader input;
+    private PrintWriter output;
+    private PrintWriter error;
+    private boolean stopRequested = false;
+    private boolean interactiveMode = true;
+    private Object lastResult = null;
+    
+    private static class NullWriter
+    extends Writer
+    {
+        private Writer parentWriter;
+
+        public NullWriter()
+        {
+            super();
+        }
+
+        public NullWriter(Object lock)
+        {
+            super(lock);
+        }
+
+        public NullWriter(Writer parentWriter)
+        {
+            super();
+            this.parentWriter = parentWriter;
+        }
+
+        public Writer getParentWriter()
+        {
+            return parentWriter;
+        }
+
+        public void write(int c) throws IOException
+        {
+        }
+
+        public void write(char cbuf[]) throws IOException
+        {
+        }
+
+        public void write(String str) throws IOException
+        {
+        }
+
+        public void write(String str, int off, int len) throws IOException
+        {
+        }
+
+        public Writer append(CharSequence csq) throws IOException
+        {
+            return this;
+        }
+
+        public Writer append(CharSequence csq, int start, int end) throws IOException
+        {
+            return this;
+        }
+
+        public Writer append(char c) throws IOException
+        {
+            return this;
+        }
+
+        public void write(char cbuf[], int off, int len) throws IOException
+        {
+        }
+
+        public void flush() throws IOException
+        {
+        }
+
+        public void close() throws IOException
+        {
+        }
+    }
+
+    public ReplEngine()
+    {
+        context = new BasicContext();
+        eval = new Eval(context);
+        eval.setCommandRepo(extensions.getCommandRepository());
+        eval.setMacroRepo(extensions.getMacroRepository());
+    }
+    
+    private void recover()
+    {
+        // Clear the current command, since we are done with it.
+        currCmd.setLength(0);
+        // Change the command prompt to normal.
+        setPrompt(PROMPT_NORMAL);
+    }
+    
+    protected void handleLine(String aLine)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 162-181 into a separate method.